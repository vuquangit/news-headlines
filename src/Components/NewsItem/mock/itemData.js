export const mockData = {
  source: {
    id: "the-irish-times",
    name: "The Irish Times",
  },
  author: "The Irish Times",
  title:
    "LIVE Coronavirus: 4,500 tests to be carried out per day from this week, HSE says",
  description:
    "70,000 applications made to Ireland on Call healthcare worker scheme",
  url:
    "https://www.irishtimes.com/news/ireland/irish-news/coronavirus-4-500-tests-to-be-carried-out-per-day-from-this-week-hse-says-1.4221258",
  urlToImage:
    "https://www.irishtimes.com/polopoly_fs/1.4221277.1586082453!/image/image.jpg",
  publishedAt: "2020-04-05T10:53:21.0103778Z",
  content:
    "A medic wearing personal protective equipment in Bergamo, Italy. Some of the consignment of PPE to Ireland last weekend was found by the HSE to be not suitable for use. Photograph: Getty",
};
