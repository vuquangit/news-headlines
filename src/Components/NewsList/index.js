import React from "react";
import { isEmpty } from "lodash";
import PropTypes from "prop-types";

import NewsItem from "Components/NewsItem";
import "./newsList.scss";

const NewsList = ({ items = [] }) => {
  return (
    <div className="news-list">
      {!isEmpty(items) &&
        items.map((item, idx) =>
          item.title ? <NewsItem key={idx} {...item} /> : null
        )}
    </div>
  );
};

NewsList.propTypes = {
  items: PropTypes.array,
};

export default NewsList;
