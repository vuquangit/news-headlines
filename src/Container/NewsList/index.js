import React from "react";
import { isEmpty } from "lodash";

import NewsItem from "Container/NewsItem";
import "./newsList.scss";

const NewsList = ({ items = [] }) => {
  const _renderItem = () =>
    !isEmpty(items) &&
    items.length > 0 &&
    items.map((item, idx) => <NewsItem key={idx} {...item} />);

  return <div className="news-list">{_renderItem()}</div>;
};

export default NewsList;
