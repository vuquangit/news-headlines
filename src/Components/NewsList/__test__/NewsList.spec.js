import React from "react";
import { render } from "@testing-library/react";

import NewsList from "..";
import { mockData } from "../mock/newsList";

describe("News List", () => {
  it("Test get default NewsList no params ", () => {
    // test snapshot
    const wrapper = render(<NewsList />);
    expect(wrapper).toMatchSnapshot();
  });

  it("Test get default NewsList with data ", () => {
    // test snapshot
    const wrapper = render(<NewsList items={mockData} />);
    expect(wrapper).toMatchSnapshot();
  });
});
