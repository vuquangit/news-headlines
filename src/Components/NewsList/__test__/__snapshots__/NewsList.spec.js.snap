// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`News List Test get default NewsList no params  1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="news-list"
      />
    </div>
  </body>,
  "container": <div>
    <div
      class="news-list"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`News List Test get default NewsList with data  1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="news-list"
      >
        <div
          class="news-item"
        >
          <div
            class="news-item__img"
          >
            <div
              class="news-item__img--content"
              style="background-image: url(https://www.irishtimes.com/polopoly_fs/1.4221277.1586082453!/image/image.jpg);"
              title="LIVE Coronavirus: 4,500 tests to be carried out per day from this week, HSE says"
            />
          </div>
          <div
            class="news-item__info"
          >
            <h3
              class="news-item__info--title"
            >
              LIVE Coronavirus: 4,500 tests to be carried out per day from this week, HSE says
            </h3>
            <div>
              <span
                class="news-item__info--source"
              >
                The Irish Times
              </span>
              <span
                class="news-item__info--published-at"
              >
                2 hours ago
              </span>
            </div>
            <span
              class="news-item__info--description"
            >
              70,000 applications made to Ireland on Call healthcare worker scheme
            </span>
          </div>
        </div>
        <div
          class="news-item"
        >
          <div
            class="news-item__img"
          >
            <div
              class="news-item__img--content"
              style="background-image: url(https://ichef.bbci.co.uk/news/1024/branded_news/04B5/production/_111050210_gettyimages-1031709582.jpg);"
              title="Mast fire probe amid 5G coronavirus claims"
            />
          </div>
          <div
            class="news-item__info"
          >
            <h3
              class="news-item__info--title"
            >
              Mast fire probe amid 5G coronavirus claims
            </h3>
            <div>
              <span
                class="news-item__info--source"
              >
                Hacker News
              </span>
              <span
                class="news-item__info--published-at"
              >
                2 hours ago
              </span>
            </div>
            <span
              class="news-item__info--description"
            >
              There have been fires at masts in Birmingham, Liverpool and Melling in Merseyside.
            </span>
          </div>
        </div>
        <div
          class="news-item"
        >
          <div
            class="news-item__img"
          >
            <div
              class="news-item__img--content"
              style="background-image: url(https://avatars2.githubusercontent.com/u/1259448?s=400&v=4);"
              title="qubole/afctl"
            />
          </div>
          <div
            class="news-item__info"
          >
            <h3
              class="news-item__info--title"
            >
              qubole/afctl
            </h3>
            <div>
              <span
                class="news-item__info--source"
              >
                Hacker News
              </span>
              <span
                class="news-item__info--published-at"
              >
                2 hours ago
              </span>
            </div>
            <span
              class="news-item__info--description"
            >
              afctl helps to manage and deploy Apache Airflow projects faster and smoother. - qubole/afctl
            </span>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="news-list"
    >
      <div
        class="news-item"
      >
        <div
          class="news-item__img"
        >
          <div
            class="news-item__img--content"
            style="background-image: url(https://www.irishtimes.com/polopoly_fs/1.4221277.1586082453!/image/image.jpg);"
            title="LIVE Coronavirus: 4,500 tests to be carried out per day from this week, HSE says"
          />
        </div>
        <div
          class="news-item__info"
        >
          <h3
            class="news-item__info--title"
          >
            LIVE Coronavirus: 4,500 tests to be carried out per day from this week, HSE says
          </h3>
          <div>
            <span
              class="news-item__info--source"
            >
              The Irish Times
            </span>
            <span
              class="news-item__info--published-at"
            >
              2 hours ago
            </span>
          </div>
          <span
            class="news-item__info--description"
          >
            70,000 applications made to Ireland on Call healthcare worker scheme
          </span>
        </div>
      </div>
      <div
        class="news-item"
      >
        <div
          class="news-item__img"
        >
          <div
            class="news-item__img--content"
            style="background-image: url(https://ichef.bbci.co.uk/news/1024/branded_news/04B5/production/_111050210_gettyimages-1031709582.jpg);"
            title="Mast fire probe amid 5G coronavirus claims"
          />
        </div>
        <div
          class="news-item__info"
        >
          <h3
            class="news-item__info--title"
          >
            Mast fire probe amid 5G coronavirus claims
          </h3>
          <div>
            <span
              class="news-item__info--source"
            >
              Hacker News
            </span>
            <span
              class="news-item__info--published-at"
            >
              2 hours ago
            </span>
          </div>
          <span
            class="news-item__info--description"
          >
            There have been fires at masts in Birmingham, Liverpool and Melling in Merseyside.
          </span>
        </div>
      </div>
      <div
        class="news-item"
      >
        <div
          class="news-item__img"
        >
          <div
            class="news-item__img--content"
            style="background-image: url(https://avatars2.githubusercontent.com/u/1259448?s=400&v=4);"
            title="qubole/afctl"
          />
        </div>
        <div
          class="news-item__info"
        >
          <h3
            class="news-item__info--title"
          >
            qubole/afctl
          </h3>
          <div>
            <span
              class="news-item__info--source"
            >
              Hacker News
            </span>
            <span
              class="news-item__info--published-at"
            >
              2 hours ago
            </span>
          </div>
          <span
            class="news-item__info--description"
          >
            afctl helps to manage and deploy Apache Airflow projects faster and smoother. - qubole/afctl
          </span>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
